File: 06Adapter/Students/2018/MielewczykLukasz/adapterSeabird/Aircraft.cs
namespace AdapterSeabird
{
    public class Aircraft : IAircraft
    {
        int height;
        bool airborne;

        public Aircraft()
        {
            height = 0;
            airborne = false;
        }

        public virtual void TakeOff()
        {
            airborne = true;
            height = 200;
        }

        public virtual bool Airborne
        {
            get { return airborne; }
        }

        public int Height
        {
            get { return height; }
            set { height = value; }
        }
    }
}

File: 06Adapter/Students/2018/kkorszun/TwoWayAdapter/src/main/scala/korszun/kacper/Aircraft.scala
package korszun.kacper

class Aircraft extends IAircraft {
  private var height = 0
  private var airborne: Boolean = false
  override def Airborne: Boolean = airborne

  override def TakeOff: Unit = {
    println("Aircraft engine takeoff")
    airborne = true
    height = 200; // Meters
  }

  override def Height: Int = height
}
